test/AsmResolver.Workspaces.DotNet.Tests/AssemblyTraversalTest.cs

```diff
@@ -0,0 +1,268 @@
+
+namespace AsmResolver.Workspaces.DotNet.Tests
+{
+    public class AssemblyTraversalTest : IClassFixture<TestCasesFixture>
+    {
+        private readonly TestCasesFixture _fixture;
+
+        public AssemblyTraversalTest(TestCasesFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Fact]
+        public void ModuleCheck()
+        {
+            var moduleDefinitions = GetAllMembers<ModuleDefinition>(_fixture.AllAssemblies, TableIndex.Module);
+            TraverseObjects(moduleDefinitions);
+        }
+
+        [Fact]
+        public void TypeRefCheck()
+        {
+            var typeReferences = GetAllMembers<TypeReference>(_fixture.AllAssemblies, TableIndex.TypeRef)
+                .Where(tr=> !tr.Name.Contains("KeyValuePair")); //CustomAttributesAssembly have redundant type reference
+            TraverseObjects(typeReferences);
+        }
+
+        [Fact]
+        public void TypeDefCheck()
+        {
+            var typeDefinitions = GetAllMembers<TypeDefinition>(_fixture.AllAssemblies, TableIndex.TypeDef);
+            TraverseObjects(typeDefinitions);
+        }
+
+        [Fact]
+        public void FieldCheck()
+        {
+            var fieldDefinitions = GetAllMembers<FieldDefinition>(_fixture.AllAssemblies, TableIndex.Field);
+            TraverseObjects(fieldDefinitions);
+        }
+
+        [Fact]
+        public void MethodCheck()
+        {
+            var methodDefinitions = GetAllMembers<MethodDefinition>(_fixture.AllAssemblies, TableIndex.Method);
+            TraverseObjects(methodDefinitions);
+        }
+
+        [Fact]
+        public void ParamCheck()
+        {
+            var parameterDefinitions = GetAllMembers<ParameterDefinition>(_fixture.AllAssemblies, TableIndex.Param);
+            TraverseObjects(parameterDefinitions);
+        }
+
+        [Fact]
+        public void InterfaceImplCheck()
+        {
+            var interfaceImplementations = GetAllMembers<InterfaceImplementation>(_fixture.AllAssemblies, TableIndex.InterfaceImpl);
+            TraverseObjects(interfaceImplementations);
+        }
+
+        [Fact]
+        public void MemberRefCheck()
+        {
+            var memberReferences = GetAllMembers<MemberReference>(_fixture.AllAssemblies, TableIndex.MemberRef);
+            TraverseObjects(memberReferences);
+        }
+
+        [Fact]
+        public void CustomAttributeCheck()
+        {
+            var customAttributes = GetAllMembers<CustomAttribute>(_fixture.AllAssemblies, TableIndex.CustomAttribute);
+            TraverseObjects(customAttributes);
+        }
+
+        [Fact]
+        public void DeclSecurityCheck()
+        {
+            var securityDeclarations = GetAllMembers<SecurityDeclaration>(_fixture.AllAssemblies, TableIndex.DeclSecurity);
+            TraverseObjects(securityDeclarations);
+        }
+
+
+        //TODO: Some standalone signatures are used for assembly building but not afterwards so some filtering is required
```

Comment:
Not sure what this means ?

Reply:
CilRawFatMethodBody contains StandaloneSignature which is not instanced by Assembly itself, rather it's used to create  LocalVariables inside CilMethodBody.
And because of that, there is no way for me to traverse them thru normal means.
This would mean I would have to implement inside ModuleAnalyzer similar method to gather all StandaloneSignatures and schedule them